java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -83,48 +120,65 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
     }
 
     /**
-     * Sends the provided [ProxyMessage] to the store backing the provided [referenceId].
+     * Sends the provided [ProxyMessage] to the store backing the provided [muxId].
      *
-     * A new store will be created for the [referenceId], if necessary.
+     * A new store will be created for the [muxId], if necessary.
      */
     suspend fun onProxyMessage(
-        message: ProxyMessage<Data, Op, T>,
-        referenceId: String
+        message: ProxyMessage<Data, Op, T>
     ): Boolean {
-        val (id, store) = store(referenceId)
-        val deMuxedMessage: ProxyMessage<Data, Op, T> = when (message) {
-            is SyncRequest -> SyncRequest(id)
-            is ModelUpdate -> ModelUpdate(message.model, id)
-            is Operations -> if (message.operations.isNotEmpty()) {
-                Operations(message.operations, id)
-            } else return true
+        val muxId = requireNotNull(message.muxId) {
+            "messages sent to Direct Store Muxer must have a muxId"
         }
-        return store.onProxyMessage(deMuxedMessage)
+        val (_, store) = getStore(muxId, message.id!!)
+        return store.onProxyMessage(message)
     }
 
-    /* internal */ suspend fun setupStore(referenceId: String): StoreRecord<Data, Op, T> {
+    /* internal */ suspend fun setupStore(muxId: String): StoreRecord<Data, Op, T> {
         val store = DirectStore.create<Data, Op, T>(
             StoreOptions(
-                storageKey = storageKey.childKeyWithComponent(referenceId),
+                storageKey = storageKey.childKeyWithComponent(muxId),
                 type = backingType
             )
         )
 
-        val id = store.on(callbackFactory(referenceId))
-
-        // Return a new Record and add it to our local stores, keyed by muxId.
-        return StoreRecord(id, store)
+        // Return a new Record with initially empty idSet
+        return StoreRecord(mutableSetOf<Int>(), store)
     }
 
-    @VisibleForTesting
-    suspend fun store(id: String) = storeMutex.withLock {
-        stores.getOrPut(id) {
-            setupStore(id)
+    /**
+     * Return the [StoreRecord] for a given muxId.
+     *
+     * If an id is provided, ensures that the id is registered to the [Store] of the [StoreRecord]
+     */
+    suspend fun getStore(muxId: String, id: Int? = null): StoreRecord<Data, Op, T> {
```

Comment:
Does this need to be public?

Reply:
I'd like to be able to use this in tests too

Reply:
you can say in a comment that it is visible only for tests, but please do not use otherwise